src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -48,33 +58,96 @@ public void run(Connection conn, List<Object> params) throws SQLException {
         String[] paramsTypes = queryTemplateInfo.getParamsTypes();
+        int j = 0;
         for (int i = 0; i < paramsTypes.length; i++) {
             if (paramsTypes[i].equalsIgnoreCase("NULL")) {
                 stmt.setNull(i + 1, Types.NULL);
+            } else if (paramsTypes[i].equalsIgnoreCase("RANDOM")) {
+                int index = i + j;
+                String distType = params.get(index).toString();
+                int minI, maxI;
+                switch (distType) {
+                    case "zipf":
+                        ZipfianGenerator zipfGen = (ZipfianGenerator) randomGenerators.get(index);
+                        stmt.setInt(i + 1, zipfGen.nextInt());
+                        j += 2;
+                        break;
+                    case "uniform":
+                        minI = Integer.parseInt(params.get(index + 1).toString());
+                        maxI = Integer.parseInt(params.get(index + 2).toString());
+                        Random rnd = (Random) randomGenerators.get(index);
+                        int uVal = rnd.nextInt(maxI - minI) + minI;
+                        stmt.setInt(i + 1, uVal);
+                        j += 2;
+                        break;
+                    case "binomial":
+                        minI = Integer.parseInt(params.get(index + 1).toString());
+                        maxI = Integer.parseInt(params.get(index + 2).toString());
+                        Random rng = (Random) randomGenerators.get(index);
+                        int bVal;
+                        do {
+                            bVal = (int) (minI + Math.abs(rng.nextGaussian()) * maxI);
+                        } while (bVal > maxI || bVal < minI);
+
+                        stmt.setInt(i + 1, bVal);
+                        j += 2;
+                        break;
+                    case "scrambled":
+                        ScrambledZipfianGenerator scramZipf = (ScrambledZipfianGenerator) randomGenerators.get(index);
+                        stmt.setInt(i + 1, scramZipf.nextInt());
+                        j += 2;
+                        break;
+                    case "string":
+                        maxI = Integer.parseInt(params.get(index + 1).toString());
+                        Random rand = (Random) randomGenerators.get(index);
+                        String randText = TextGenerator.randomStr(rand, maxI);
+                        stmt.setString(i + 1, randText);
+                        j += 1;
+                        break;
+                    case "datetime":
+                    case "timestamp":
+                        stmt.setTimestamp(i + 1, new Timestamp(System.currentTimeMillis()));
+                        break;
+                    case "date":
+                        stmt.setDate(i + 1, new Date(System.currentTimeMillis()));
+                        break;
+                    case "time":
+                        stmt.setTime(i + 1, new Time(System.currentTimeMillis()));
```

Comment:
These time types aren't actually `RANDOM`, right?
Again, it'd be useful to be able to specify random time ranges for many benchmarks.